CMakeCommon.cmake

```diff
@@ -246,3 +251,42 @@ function (GenerateAddOnProject target acVersion devKitDir addOnName addOnSources
     SetCompilerOptions (${target} ${acVersion})
 
 endfunction ()
+
+function (check_valid_language_code configFile languageCode)
+    file (READ ${configFile} configsContent)
+    string (JSON configuredLanguagesList GET "${configsContent}" "languages")
+    string (JSON configuredLanguagesListLen LENGTH "${configsContent}" "languages")
+    set (i 0)
+    while (${i} LESS ${configuredLanguagesListLen})
```

Comment:
The [conditional syntax of `LESS`](https://cmake.org/cmake/help/latest/command/if.html#less) is `<variable|string> LESS <variable|string>`. Variable substitution happens before conditional evaluation. This means that the replacement is first observed as a variable and only then as a string. To avoid this, just name variable(s) directly without quoting: `while (i LESS configuredLanguagesListLen)`

Please apply to the other instances within the additions as well.